web3.js/babel.config.js

```diff
@@ -0,0 +1,62 @@
+const isCalledByBundler = caller => caller.name.startsWith('@rollup');
+
+module.exports = api => {
+  const calledByBundler = api.caller(isCalledByBundler);
+  const transformEnvVariables =
+    !calledByBundler &&
+    process.env.NODE_ENV &&
+    process.env.BROWSER !== undefined;
+
+  return {
+    presets: [
+      [
+        '@babel/preset-env',
+        {
+          modules: false,
+          bugfixes: true,
+        },
+      ],
+      ['@babel/preset-typescript'],
+    ],
+    plugins: [
+      [
+        '@babel/plugin-proposal-class-properties',
+        {
+          loose: true,
+        },
+      ],
+      [
+        '@babel/plugin-proposal-private-methods',
+        {
+          loose: true,
+        },
+      ],
+      [
+        '@babel/plugin-proposal-private-property-in-object',
+        {
+          loose: true,
+        },
+      ],
+      transformEnvVariables && [
+        'transform-inline-environment-variables',
+        {
+          include: ['NODE_ENV', 'BROWSER'],
+        },
+      ],
+      transformEnvVariables && [
+        'minify-dead-code-elimination',
+        {
+          keepFnName: true,
+          keepFnArgs: true,
+          keepClassName: true,
+        },
+      ],
+      transformEnvVariables && [
+        'babel-plugin-transform-remove-imports',
+        {
+          test: 'agent-manager',
```

Comment:
Not really satisfied by this solution, forces the build to be aware that a specific import is unused in specific conditions ̛& needs to be removed.

I believe it might be best to not remove the import, and in fact ship the code containing references to `http[s]` Node.js module, even if it forces additional yet trivial configuration of consumers (which in most cases, is covered by frameworks build)